include/roboteam_utils/Line.h

```diff
@@ -171,37 +177,7 @@ class Line {
      * @return true if \ref line intersects `this`
      * @return false False if \ref line does not intersect `this`
      */
-    [[nodiscard]] bool doesIntersect(const Line &line) const;
-
-    /**
-     * @brief Checks whether \ref line intersects `this`
-     *
-     * @param line Line to check against
-     * @return true True if \ref line intersects `this`
-     * @return false False if \ref line does not intersect `this`
-     */
-    [[nodiscard]] bool doesIntersect(const LineSegment &line) const;
-
-    /**
-     * Get the intersection point between two infinite lines. No intersection point is returned in case the lines are equal or parallel.
-     *
-     * @param line1Start An arbitrary point on the first line.
-     * @param line1End An arbitrary other point on the first line (make sure that it is different than line1Start).
-     * @param line2Start An arbitrary point on the second line.
-     * @param line2End An arbitrary other point on the second line (make sure that it is different than line2Start).
-     * @return std::nullopt if the lines do not intersect or are equal. Otherwise return the intersection point.
-     */
-    static std::optional<Vector2> intersect(const Vector2 line1Start, const Vector2 line1End, const Vector2 line2Start, const Vector2 line2End);
-
-    /**
-     * Get the relative position of pointOnLine on the given infinite line, i.e. compute t such that line1Start + (line1End - line1Start) * t = pointOnLine
-     *
-     * @param line1Start An arbitrary point on the first line.
-     * @param line1End An arbitrary other point on the first line (make sure that it is different than line1Start).
-     * @param pointOnLine A point that is located on the given line.
-     * @return The value t such that line1Start + (line1End - line1Start) * t = pointOnLine
-     */
-    static float relativePosition(const Vector2 line1Start, const Vector2 line1End, const Vector2 pointOnLine);
```

Comment:
Why are these removed?

Reply:
The last 2 methods are not removed, but their definitions have been changed. The doesIntersect methods have been removed, because the Line and LineSegment classes have totally been redesigned and I do not spend time redesigning dead methods.